frontend/src/components/SystemSubmitDrawer/index.tsx

```diff
@@ -237,6 +249,16 @@ export function SystemSubmitDrawer(props: Props) {
         }
       }
     }
+    if (changedFields.source_language != null) {
```

Comment:
If you specify **label -> name** and **value -> language_code** for `<Select />`, it will automatically `setFieldsValue` for you.